handler.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+var AWS = require('aws-sdk');
+var attr = require('dynamodb-data-types').AttributeValue;
+var craigslistApartments = require('./craigslist-apartments');
+var filters = require('./filters');
+
+
+var ddb = new AWS.DynamoDB({apiVersion: '2012-08-10'});
+
+module.exports.hello = (event, context) => {
+  const filter = new filters(event.filters.postalCode, event.filters.kmFromPostalCode, event.filters.minPrice, event.filters.maxPrice, event.filters.minSqft);
+  var apartmentsPromise = craigslistApartments.getListOfRentals(filter);
+  return apartmentsPromise.then((result) => {
+    return result;
+  })
+  .then((apartmentListings) => {
+    return storeInDB(ddb, apartmentListings);
+  })
+  .then((todaysListings) => {
+    var yesterdayListingsPromise = getYesterdaysListings(ddb);
+    return yesterdayListingsPromise.then((result) => {
+      return result;
+    })
+    .then((yesterdayListings) => {
+      var apartmentListings = {
+        todaysListings: todaysListings,
+        yesterdayListings: yesterdayListings
+      }
+      return apartmentListings;
+    })
+    
+  })
+  .then((apartmentListings) => {
+    var apartments = findNewApartments(apartmentListings);
+    var removed = findRemovedApartments(apartmentListings);
+    var message = createdMessage(apartments, removed);
+    return sendEmail(message);
+  })
+  .then(() => {
+    return {
+      statusCode: 200,
+      body: JSON.stringify({ message: "Lambda executed with success" })
+    }
+  })
+}
+
+function findNewApartments(apartmentListings) {
+  var apartments = {
+    newApartments: [],
+    oldApartments: []
+  };
+  apartmentListings.todaysListings.forEach(function(element) {
+    var elementPos = apartmentListings.yesterdayListings.listings.map(function(x) {return x.id; }).indexOf(element.id); 
+    if(elementPos === -1) {
+      apartments.newApartments.push(element);
+    } else {
+      apartments.oldApartments.push(element);
+    }
+  });
+  return apartments;
+}
+
+function findRemovedApartments(apartmentListings) {
+  var removedApartments = [];
+  apartmentListings.yesterdayListings.listings.forEach(function(element) {
+    var elementPos = apartmentListings.todaysListings.map(function(x) {return x.id; }).indexOf(element.id); 
+    if(elementPos === -1) {
+      removedApartments.push(element);
+    } 
+  });
+  return removedApartments;
+}
+
+function storeInDB(ddb, apartmentListings) {
+  var today = new Date();
+  var data = {
+    site: 'Craigslist',
+    date: getFormattedDate(today),
+    listings: apartmentListings,
+    updated: today.toString()
+  }
+
+  var dynamodbData = attr.wrap(data);
+  
+  var params = {
+    TableName: 'rentals1table',
```

Comment:
Why is the table name hardcoded? In your serverless.yml you set the environment variable `tableName`. In this line you have to use this env variable.

Reply:
Done.